lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1065,6 +1117,22 @@ def exec_module(self, **kwargs):
 
                         vm_resource.storage_profile.data_disks = data_disks
 
+                    # Before creating VM accept terms of plan if `accept_terms` is True
+                    if self.accept_terms:
+                        if not self.plan:
+                            self.fail("parameter error: plan must be specified and include name, product, and publisher")
+                        try:
+                            plan_name = self.plan.get('name')
+                            plan_product = self.plan.get('product')
+                            plan_publisher = self.plan.get('publisher')
+                            term = self.marketplace_client.marketplace_agreements.get(
+                                plan_publisher, plan_product, plan_name)
+                            term.accepted = True
+                            agreement = self.marketplace_client.marketplace_agreements.create(
```

Comment:
where to use this parameter? I didn't find it..

Reply:
This is the block of code which takes the optional `accept_terms` parameter specified in https://github.com/ansible/ansible/pull/43695/files#diff-7deb317ec8158f70119f43a48d9ba545R723 and uses this to accept the market place terms.
It is _used_ in that Azure has a record that the terms have been accepted for a Marketplace image, such that when the VM is provisioned a few moments later on `create_or_update` it will continue without a hitch. Without this VM creation fails requiring accepting of the terms to be done in some (generally manual) way.